File: apps/backend/src/main/java/com/supertech/backend/plan/service/PlanService.java
package com.supertech.backend.plan.service;

import java.util.List;

import com.supertech.backend.plan.dto.CreatePlanRequest;
import com.supertech.backend.plan.dto.PlanResponse;
import com.supertech.backend.plan.dto.UpdatePlanRequest;

public interface PlanService {
    void create(CreatePlanRequest requset);

    void update(UpdatePlanRequest requset, Long id);

    void delete(Long id);

    List<PlanResponse> getAll();

    PlanResponse getById(Long id);
}

File: apps/frontend/my-app/src/app/features/plan/service/plan.service.ts
import api from "@/app/common/lib/axios";
import { ApiResponse } from "@/app/common/types/api.types";
import {
  CreatePlanRequest,
  PlanResponse,
  UpdatePlanRequest,
} from "../types/plan.types";

export const getAll = async () => {
  const res = await api.get<ApiResponse<PlanResponse[]>>("/plan");
  return res.data;
};

export const getById = async (id: number) => {
  const res = await api.get<ApiResponse<PlanResponse>>(`/plan/${id}`);
  return res.data;
};

export const create = async (data: CreatePlanRequest) => {
  const res = await api.post<ApiResponse<void>>("/plan", data);
  return res.data;
};

export const update = async (id: number, data: UpdatePlanRequest) => {
  const res = await api.put<ApiResponse<void>>(`/plan${id}`, data);
  return res.data;
};

export const remove = async (id: number) => {
  const res = await api.delete<ApiResponse<void>>(`/plan/${id}`);
  return res.data;
};
